src/dmd/e2ir.d

```diff
@@ -3531,6 +3531,18 @@ elem *toElem(Expression e, IRState *irs)
 
         override void visit(CondExp ce)
         {
+            if (auto ve = ce.econd.isVarExp)
+                if (ve.var.ident == Id.ctfe)
+                {
+                    elem *e = toElem(ce.e2, irs);
+                    if (irs.params.cov && ce.e2.loc.linnum)
+                        e = el_combine(incUsageElem(irs, ce.e2.loc), e);
+                    if (tybasic(e.Ety) == TYstruct)
+                        e.ET = Type_toCtype(ce.e2.type);
+                    elem_setLoc(e, ce.loc);
+                    result = e;
+                    return;
+                }
             elem *ec = toElem(ce.econd, irs);
```

Comment:
Seems like a good idea to me. Needs to be in a different PR though and I guess there's already a bug report for this somewhere.

Ideally, this should not hard-detect the ctfe variable though. You code will fail on constructs such as these:
```d
void foo(bool dummy)
{
    if (__ctfe && dummy)
    {
        // CTFE-only Code goes here
    }
}
```
What should be done instead is this: Set `__ctfe` (somehow) to false, then constant fold the condition. If the result is a constant `false`, the if part of the condition is a ctfe only part. If it is constant true, the `else` branch is a ctfe only branch. If it is not const, both branches may be used at runtime.

Maybe that's out of scope for GSoC though, but it would be very useful for betterC.